src/map/lua/lua_baseentity.cpp

```diff
@@ -366,6 +366,106 @@ inline int32 CLuaBaseEntity::isJugPet(lua_State* L)
 
 //======================================================//
 
+inline int32 CLuaBaseEntity::getCharmChance(lua_State* L)
```

Comment:
Not quite; I used some of the code from both "tryToCharm" and "TryCharm" in battleutils.cpp to get the charm percentage chance.
Those functions actually try to charm a mob, so I needed a function that just gave me the chance the player had to charm a particular target.